### PR TITLE
Publicly expose Coregistration instance from CoregistrationUI

### LINKED ISSUE #10227: Retrieving Coregistration from CoregistrationUI
#### Describe the new feature or enhancement
I need to programmatically retrieve the coregistration results from `CoregistrationUI` after the user has finished coregistration, without requiring them to explicitly save `trans` and `fiducials` to disk.

Currently, there's no public API to do so: the `Coregistration` instance is stored in the private `CoregistrationUI._coreg` – which is overwritten with `None` when the `CoregistrationUI` window is closed.

#### Describe your proposed implementation
`CoregistrationUI._coreg` should become public as `CoregistrationUI.coreg` or `CoregistrationUI.coregistration`, and `CoregistrationUI._clean()` should not overwrite the `Coregistration` instance on window close.

#### Describe possible alternatives
`mne.gui.coregistration()` could return the `Coregistration` instance after `CoregistrationUI` has been closed, instead of returning `CoregistrationUI`. But I believe this would break some workflows. Or wouldn't even be possible in non-blocking mode…

cc @GuillaumeFavelier @larsoner

### PR DESCRIPTION
**Pull request by hoechenberger**
This exposes `Coregistration` via `CoregistrationUI.coreg`

Closes #10227

### CODE REVIEW

File: mne/gui/_coreg.py [hunk 1]
```diff
@@ -90,8 +90,13 @@ class CoregistrationUI(HasTraits):
...
         .. versionadded:: 1.0
     %(verbose)s
-    """
 
+    Attributes
+    ----------
+    coreg : mne.coreg.Coregistration
+        The coregistration instance used by the graphical interface.
+    """
+
```
Comment (hoechenberger) [inline mne/gui/_coreg.py:99]:
```suggestion
```

**Comment by hoechenberger** [comment]
(I'm having trouble with the doc build locally, so will need to see whether Circle comes back green here. Otherwise this is good to go from my end)

**Comment by hoechenberger** [comment]
CIs are having issues with downloading stuff, I'm restarting checks

**Comment by larsoner** [comment]
```
mne/gui/_coreg.py:43 in public class `CoregistrationUI`:
        D204: 1 blank line required after class docstring (found 0)

```

**Comment by hoechenberger** [comment]
> D204: 1 blank line required after class docstring (found 0)

Urgh, I specifically REMOVED the blank line there … sigh

**Comment by larsoner** [comment]
Thanks @hoechenberger !

**Pull request merged**